database/postgres/save_data.py

```diff
@@ -87,8 +99,21 @@ def save_api_data(json_data: Dict, sport_name: str) -> None:
                     date = None
 
 
-                if not isinstance(status, str) and isinstance(status, dict):
-                    status = status.get('long')
+                if isinstance(status, dict):
+                    status = status.get('long', '')
+
+                if not isinstance(status, str) or not status:
+                    status = IN_PROGRESS
+
+                status_lower = status.lower()
```

Comment:
```suggestion
                status_lower = status.lower()
                game_type_result = None
                
                for game_type, phrases in CUSTOM_GAMES.items():
                    if any(phrase in status_lower for phrase in map(str.lower, phrases)):  
                        game_type_result = game_type
                        break

```

1. Avoid using `type` in variable naming, since this word is reserved by python
2. Preprocess all phrases at once instead of doing .lower() for each separately